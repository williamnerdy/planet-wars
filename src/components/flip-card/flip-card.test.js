import React from 'react';
import { shallow } from 'enzyme';
import FlipCard from './flip-card';

describe('<FlipCard />', () => {
  let helloWorld = null;

  beforeEach(() => {
    helloWorld = <h3>Hello World!</h3>;
  });

  it('should render correctly', () => {
    const element = shallow(<FlipCard />);
    expect(element).toMatchSnapshot();
  });

  it('should render correctly with "Hello World!" on front', () => {
    const element = shallow(<FlipCard front={helloWorld} />);
    expect(element).toMatchSnapshot();
  });

  it('should have "Hello World!" on front', () => {
    const element = shallow(<FlipCard front={helloWorld} />);
    expect(element.contains(helloWorld)).toBe(true);
  });

  it('should render correctly with "Hello World!" on back', () => {
    const element = shallow(<FlipCard back={helloWorld} />);
    expect(element).toMatchSnapshot();
  });

  it('should have "Hello World!" on back', () => {
    const element = shallow(<FlipCard back={helloWorld} />);
    expect(element.contains(helloWorld)).toBe(true);
  });
});
