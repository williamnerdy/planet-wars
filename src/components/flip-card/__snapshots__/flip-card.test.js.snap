// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FlipCard /> should render correctly 1`] = `ShallowWrapper {}`;

exports[`<FlipCard /> should render correctly with "Hello World!" on back 1`] = `ShallowWrapper {}`;

exports[`<FlipCard /> should render correctly with "Hello World!" on front 1`] = `ShallowWrapper {}`;
